api

```diff
@@ -687,7 +692,7 @@ Package: $package_name" > /tmp/$package_name/DEBIAN/control
     done
     
     errors="$(echo "$output" | grep '^[(E)|(Err]:')"
-    if [ ! -z "$errors" ];then
+    if [ ! -z "$errors" ] || [ "$exitcode" != 0 ];then
```

Comment:
this will catch the issue but the debug output will not make sense.

it would be better to add an `else` case below this with custom output indicating that apt exited with an error without any `E` or `Err` output.

Reply:
In this if statement, other specific messages from apt are checked for. Splitting this into an elif would mean either skipping those checks or duplicating them. Is that worth it?

Reply:
then simply check for the exitcode again inside if statement along with the other cases to avoid needing to duplicate them.

APT only has `0` and `100` exit codes programmed into it (manpage says so https://manpages.ubuntu.com/manpages/noble/man8/apt.8.html) . If it exits with something no equaling one of those two values it is an undefined error (like a segfault).